src/librustc_typeck/diagnostics.rs

```diff
@@ -1915,6 +1915,47 @@ More details can be found in [RFC 438].
 [RFC 438]: https://github.com/rust-lang/rfcs/pull/438
 "##,
 
+E0182: r##"
+You bound an associated type in an expression path which is not
+allowed. Erroneous code example:
+
+```compile_fail,E0182
+trait Foo {
+    type A;
+    fn bar() -> isize;
+}
+
+impl Foo for isize {
+    type A = usize;
+    fn bar() -> isize { 42 }
+}
+
+pub fn main() {
+    // error: unexpected binding of associated item in expression path
+    let x: isize = Foo::<A=usize>::bar();
+}
+```
+
+To give a concrete type when using the Universal Function Call Syntax,
```

Comment:
I wonder if there are enough details or not... Hard to say from my point of view... :-/

What do you think of this @steveklabnik ?